### PR TITLE
Added Capp status sync addon.

### PR DESCRIPTION
**Pull request by fran2810**
Added two components:
manger -  responsible on the life cycle of agents on the managed cluster. agent -responsible to update the Capp statuses on the hub cluster. The agent runs on the managed cluster. There is also deployments section which holds the manifests for the deployment and the deploy it self is in the make fie.

### CODE REVIEW

File: addons/README.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Capp Status Sync Add-on
```
Comment (omerbd21) [inline addons/README.md:1]: Add some explanation about the actual addon you wrote

File: addons/cmd/main.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	zapLog, err := zap.NewDevelopment()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "failed to start up logger %v\n", err)
+		os.Exit(1)
+	}
+
+	logger = zapr.NewLogger(zapLog)
+
+	command := newCommand(logger)
+	fmt.Println(command)
```
Comment (omerbd21) [inline addons/cmd/main.go:40]: Why is this here?

File: addons/hub/status_manager.go [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+	runController := func(ctx context.Context, controllerContext *controllercmd.ControllerContext) error {
+		mgr, err := addonmanager.New(controllerContext.KubeConfig)
+		if err != nil {
+			return err
+		}
+		registrationOption := newRegistrationOption(
+			controllerContext.KubeConfig,
+			controllerContext.EventRecorder,
+			componentName,
+			utilrand.String(5),
```
Comment (omerbd21) [inline addons/hub/status_manager.go:79]: move 5 to a const

File: addons/hub/status_manager.go [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+			Client:            hubClient,
+			log:               log.WithName("override-values"),
+			operatorNamespace: controllerContext.OperatorNamespace,
+			withOverride:      withOverride,
+		}
+
+		agentAddon, err := addonfactory.NewAgentAddonFactory(componentName, fs, templatePath).
+			WithGetValuesFuncs(o.getValueForAgentTemplate, addonfactory.GetValuesFromAddonAnnotation).
+			WithAgentRegistrationOption(registrationOption).
+			WithInstallStrategy(frameworkagent.InstallAllStrategy("open-cluster-management-agent-addon")).
```
Comment (omerbd21) [inline addons/hub/status_manager.go:98]: move string to const

File: addons/hub/status_manager.go [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+	for _, result := range results {
+		if result.Error != nil {
+			return result.Error
+		}
+	}
+
+	return nil
+}
+
+// getValues prepare values for templates at manifests/templates
```
Comment (omerbd21) [inline addons/hub/status_manager.go:197]: what is this comment? It refers to a nonexistent function

File: addons/hub/status_manager.go [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+func (o *override) getValueForAgentTemplate(cluster *clusterv1.ManagedCluster,
+	addon *addonapiv1alpha1.ManagedClusterAddOn) (addonfactory.Values, error) {
+	addonImage := os.Getenv("ADDON_IMAGE")
+	if len(addonImage) == 0 {
+		addonImage = "quay.io/danateamorg/champions:latest"
```
Comment (omerbd21) [inline addons/hub/status_manager.go:202]: move the img name to a const/use as parameter in makefile

File: main.go [hunk 7]
```diff
@@ -125,11 +125,13 @@ func main() {
 	}
 
 	hookServer := mgr.GetWebhookServer()
-	decoder, _ := admission.NewDecoder(scheme)
```
Comment (omerbd21) [inline main.go:128]: why did you do this change?

**Review by omerbd21 — CHANGES_REQUESTED** [review]

File: main.go (hunk 7 above)
Comment (fran2810) [reply main.go:128]: The api of controller runtime changed when we upgraded to 1.15

**Review by omerbd21 — APPROVED** [review]
/lgtm

**Comment by mzeevi** [comment]
Change every instance of `CAPP` to `Capp`. Also, the signed commit is not correct, check the user you used. @fran2810

**Review by mzeevi — APPROVED** [review]
